### ISSUE TITLE
Firebase Integration

### ISSUE THREAD

**Issue by tomoiti** [msintuneappsdk/ms-intune-app-sdk-android#25]
I followed all steps to integrate Intune to my Android project, but when I start the app I got this error activating Firebase SDK:

 java.lang.RuntimeException: Unable to get provider com.google.firebase.provider.FirebaseInitProvider: java.lang.NullPointerException: Attempt to invoke interface method 'java.lang.Object com.microsoft.intune.mam.client.ComponentsContainer.get(java.lang.Class)' on a null object reference
        at android.app.ActivityThread.installProvider(ActivityThread.java:6643)
        at android.app.ActivityThread.installContentProviders(ActivityThread.java:6185)
        at android.app.ActivityThread.handleBindApplication(ActivityThread.java:6100)
        at android.app.ActivityThread.access$1200(ActivityThread.java:213)
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1807)
        at android.os.Handler.dispatchMessage(Handler.java:106)
        at android.os.Looper.loop(Looper.java:193)
        at android.app.ActivityThread.main(ActivityThread.java:6923)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:493)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:870)
     Caused by: java.lang.NullPointerException: Attempt to invoke interface method 'java.lang.Object com.microsoft.intune.mam.client.ComponentsContainer.get(java.lang.Class)' on a null object reference
        at com.microsoft.intune.mam.client.InterfaceComponentsAccess.get(InterfaceComponentsAccess.java:30)
        at com.microsoft.intune.mam.client.CachedBehaviorProvider.get(CachedBehaviorProvider.java:33)
        at com.microsoft.intune.mam.client.content.pm.MAMPackageManagement.getBehavior(MAMPackageManagement.java:409)
        at com.microsoft.intune.mam.client.content.pm.MAMPackageManagement.getServiceInfo(MAMPackageManagement.java:298)
        at com.google.firebase.components.ComponentDiscovery$MetadataRegistrarNameRetriever.getMetadata(com.google.firebase:firebase-common@@17.1.0:131)
        at com.google.firebase.components.ComponentDiscovery$MetadataRegistrarNameRetriever.retrieve(com.google.firebase:firebase-common@@17.1.0:106)
        at com.google.firebase.components.ComponentDiscovery$MetadataRegistrarNameRetriever.retrieve(com.google.firebase:firebase-common@@17.1.0:102)
        at com.google.firebase.components.ComponentDiscovery.discover(com.google.firebase:firebase-common@@17.1.0:71)
        at com.google.firebase.FirebaseApp.<init>(com.google.firebase:firebase-common@@17.1.0:411)
        at com.google.firebase.FirebaseApp.initializeApp(com.google.firebase:firebase-common@@17.1.0:306)
        at com.google.firebase.FirebaseApp.initializeApp(com.google.firebase:firebase-common@@17.1.0:273)
        at com.google.firebase.FirebaseApp.initializeApp(com.google.firebase:firebase-common@@17.1.0:257)
        at com.google.firebase.provider.FirebaseInitProvider.onCreate(com.google.firebase:firebase-common@@17.1.0:53)
        at android.content.ContentProvider.attachInfo(ContentProvider.java:1919)
        at android.content.ContentProvider.attachInfo(ContentProvider.java:1894)
        at com.microsoft.intune.mam.client.content.MAMContentProvider.attachInfoReal(MAMContentProvider.java:142)
        at com.microsoft.intune.mam.client.app.offline.OfflineContentProviderBehavior.attachInfoMAM(OfflineContentProviderBehavior.java:151)
       at com.microsoft.intune.mam.client.content.MAMContentProvider.attachInfoMAM(MAMContentProvider.java:242)
        at com.google.firebase.provider.FirebaseInitProvider.attachInfoMAM(com.google.firebase:firebase-common@@17.1.0:47)
        at com.microsoft.intune.mam.client.app.offline.OfflineContentProviderBehavior.attachInfo(OfflineContentProviderBehavior.java:43)
        at com.microsoft.intune.mam.client.content.MAMContentProvider.attachInfo(MAMContentProvider.java:50)
        at android.app.ActivityThread.installProvider(ActivityThread.java:6638)
        at android.app.ActivityThread.installContentProviders(ActivityThread.java:6185) 
        at android.app.ActivityThread.handleBindApplication(ActivityThread.java:6100) 
        at android.app.ActivityThread.access$1200(ActivityThread.java:213) 
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1807) 
        at android.os.Handler.dispatchMessage(Handler.java:106) 
        at android.os.Looper.loop(Looper.java:193) 
        at android.app.ActivityThread.main(ActivityThread.java:6923) 
        at java.lang.reflect.Method.invoke(Native Method) 
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:493) 
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:870) 

Any clue on how to fix this issue?

**Comment by tomoiti** [comment]
Still an issue using v6.0.0

**Comment by tomoiti** [comment]
Found the issue.

I forgot to change my app to use MAMApplication (this replacement is not done by gradle plugin). It is documented in the integration docs but buried in the middle of a long series of tables that I ignored since they list code replacement that are done automatically for me by the gradle plugin.

Because of that I have two suggestions. First, change documentation to make this step even more clear and before that long series of tables. Second, improve somehow the error message.
